### PR TITLE
NEWSWORLDSERVICE-1695: Upgrade all patch & minor linting dependencies

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves https://jira.dev.bbc.co.uk/browse/NEWSWORLDSERVICE-1695

Overall changes
======
Upgrades lint packages & fixes / ignores linting errors where appropriate
It also appears that `stylelint` was not configured correctly, so lots of 'new' style linting errors have been detected and subsequently fixed.
The version of `stylelint-config-recommended` has been downgraded so that is is compatible with v13.13.1 of `stylelint`
Finally, we have disabled the `no-extra-semicolons` stylelint rule as it was causing semi-colons to be missed & snapshots to break.  

Code changes
======
Patch & minor upgrades to all linting libraries
Fixes linting errors/warnings where possible
Ignores/disables linting errors/warnings where it was not easy / feasible to fix

Testing
======
- [ ] PR Checks passing

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: .storybook/helpers/ampDecorator/index.jsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ const AmpDecorator = storyFn => (
       <link rel="canonical" href="http://foobar.com" />
       <meta charset="utf-8" />
       <meta name="viewport" content="width=device-width,minimum-scale=1" />
+      {/* eslint-disable-next-line react/no-unknown-property */}
```
Comment (karinathomasbbc) [inline .storybook/helpers/ampDecorator/index.jsx:22]: Remove this as it is already ignored in lint config

File: src/app/legacy/components/AmpGeo/index.jsx [hunk 2]
```diff
@@ -64,6 +64,7 @@ const configuration = {
 export const AMP_GEO_SCRIPT = (
   <script
     async
+    // eslint-disable-next-line react/no-unknown-property
```
Comment (karinathomasbbc) [inline src/app/legacy/components/AmpGeo/index.jsx]:
```suggestion
```

File: src/app/legacy/components/MediaPlayer/Amp/index.jsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ const AmpHead = () => (
   <Helmet>
     <script
       async
+      // eslint-disable-next-line react/no-unknown-property
```
Comment (karinathomasbbc) [inline src/app/legacy/components/MediaPlayer/Amp/index.jsx]:
```suggestion
```

File: src/app/legacy/containers/ConsentBanner/Banner/cookie.amp.jsx [hunk 4]
```diff
@@ -197,6 +198,7 @@ const AmpCookieBanner = ({
       <Helmet>
         <script
           async
+          // eslint-disable-next-line react/no-unknown-property
```
Comment (karinathomasbbc) [inline src/app/legacy/containers/ConsentBanner/Banner/cookie.amp.jsx]: Remove as it is ignored in lint config
```suggestion
```

File: .storybook/helpers/ampDecorator/index.jsx (hunk 1 above)
Comment (karinathomasbbc) [inline .storybook/helpers/ampDecorator/index.jsx]:
```suggestion
```

**Review by Isabella-Mitchell — APPROVED** [review]
🔧 ✅  🙌

File: src/app/legacy/psammead/psammead-copyright/src/index.jsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ import {
 } from '#psammead/gel-foundations/src/typography';
 
 const Copyright = styled.p`
-  ${GEL_MINION};
+  ${GEL_MINION}
```
Comment (karinathomasbbc) [inline src/app/legacy/psammead/psammead-copyright/src/index.jsx]: `;` is required here - reminder to reinstate

**Review by amoore108 — APPROVED** [review]

**Pull request merged**